pkg/controller/nodeconfig/sync_roles.go

```diff
@@ -0,0 +1,47 @@
+// Copyright (C) 2024 ScyllaDB
+
+package nodeconfig
+
+import (
+	"context"
+	"fmt"
+
+	scyllav1alpha1 "github.com/scylladb/scylla-operator/pkg/api/scylla/v1alpha1"
+	"github.com/scylladb/scylla-operator/pkg/controllerhelpers"
+	"github.com/scylladb/scylla-operator/pkg/resourceapply"
+	rbacv1 "k8s.io/api/rbac/v1"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+)
+
+func (ncc *Controller) syncRoles(ctx context.Context, nc *scyllav1alpha1.NodeConfig, roles map[string]*rbacv1.Role) error {
+	requiredRoles := []*rbacv1.Role{
+		makePerftuneRole(),
+	}
+
+	// Delete any excessive Roles.
+	// Delete has to be the first action to avoid getting stuck on quota.
+	err := controllerhelpers.Prune(
```

Comment:
shouldn't Prune complain about missing controllerref on deleted objects? We are not allowed to delete them. There should be an option to disable it on justified cases similar to apply functions.

Reply:
You already give it a set of claimed and pre-existing objects so it's assumed you own them by how you've claimed them in the fist place. If you were to do a patch or update instead you'd not have rechecked again. For the case of the helper it's cheap but I'd not couple it to this PR.

Reply:
I meant to add a safety check so we won't delete objects we weren't meant to delete. Filing an issue for future improvement is fine.

Reply:
https://github.com/scylladb/scylla-operator/issues/2010